delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala

```diff
@@ -167,9 +170,20 @@ final class ResourcesRoutes(
                       (pathPrefix("source") & get & pathEndOrSingleSlash & idSegmentRef(id)) { id =>
                         operationName(s"$prefixSegment/resources/{org}/{project}/{schema}/{id}/source") {
                           authorizeFor(ref, Read).apply {
-                            implicit val source: Printer = sourcePrinter
-                            val sourceIO                 = resources.fetch(id, ref, schemaOpt).map(_.value.source)
-                            emit(sourceIO.leftWiden[ResourceRejection].rejectWhen(wrongJsonOrNotFound))
+                            parameter("annotate".as[Boolean].withDefault(false)) { annotate =>
+                              implicit val source: Printer = sourcePrinter
+                              if (annotate) {
+                                emit(
+                                  resources
+                                    .fetch(id, ref, schemaOpt)
+                                    .flatMap(asSourceWithMetadata)
+                                )
+                              } else {
+                                val sourceIO = resources.fetch(id, ref, schemaOpt).map(_.value.source)
+                                val value    = sourceIO.leftWiden[ResourceRejection]
+                                emit(value.rejectWhen(wrongJsonOrNotFound))
+                              }
+                            }
```

Comment:
This seems like a lot of logic for a Route, but there is no layer between this and `Resources` so it seems like the best place